src/main/kotlin/io/emeraldpay/dshackle/config/UpstreamsConfig.kt

```diff
@@ -119,7 +120,24 @@ open class UpstreamsConfig {
 
     class EthereumConnection : RpcConnection() {
         var ws: WsEndpoint? = null
-        var preferHttp: Boolean = false
+        var preferHttp: Boolean? = null
+        var connectorMode: String? = null
+
+        fun resolveMode(): ConnectorMode {
+            return if (preferHttp == true) {
+                ConnectorMode.RPC_REQUESTS_WITH_MIXED_HEAD
+            } else {
+                if (connectorMode == null) {
+                    if (ws == null) {
+                        ConnectorMode.RPC_ONLY
+                    } else {
+                        ConnectorMode.WS_ONLY
+                    }
+                } else {
+                    ConnectorMode.valueOf(connectorMode!!.uppercase(Locale.getDefault()))
```

Comment:
Maybe throw an understandable exception or just print a warning/error if there is no such value?

Reply:
I see `parse` method in `ConnectorMode`, let's use it here